tmc-plugin/src/fi/helsinki/cs/tmc/utilities/urlcallback/CallbackURLStreamHandler.java

```diff
@@ -23,11 +24,11 @@
 
     private static final Map<String, URLCallback> callbacks = Maps.newHashMap();
 
-    public static void registerCallback(String path, URLCallback callback) {
+    public static void registerCallback(Path path, URLCallback callback) {
         if (callback == null) {
-            callbacks.remove(path);
+            callbacks.remove(path.toString());
         } else {
-            callbacks.put(path, callback);
+            callbacks.put(path.toString(), callback);
```

Comment:
Iff the Map is String, URLCallback, maybe it would make sense to overload this method to allow both strings and paths - or should we the map to use Paths too?

(I didn't really check for possible uses though)